mnemosyne/integrations/hermes_persona_prompt.py

```
@@ -0,0 +1,96 @@
+                return self._persona_cache["content"]
+            raw = persona_file.read_text()
+            words = raw.split()
+            max_words = int(self.PERSONA_TOKEN_CAP * 0.75)
+            if len(words) > max_words:
+                truncated = " ".join(words[:max_words])
+                last_section = truncated.rfind("\n## ")
+                if last_section > 0:
+                    truncated = truncated[:last_section]
+                raw = truncated + "\n... (truncated, see persona file for full content)"
+            self._persona_cache = {"mtime": mtime, "content": raw}
+            return raw
```

Ruff (0.15.20)

[warning] Consider moving this statement to an `else` block

(TRY300)